### PR TITLE
Misc frontend refactors and tweaks

### PR DESCRIPTION
**Pull request by vincerubinetti**
- fix unit of Public Water System Violations
- use `cent_lat` and `cent_long` that have apparently always been in the backend, instead of using turf.js to calc center of mass for label positions
- rearrange api funcs a bit
- ~move `index` -> `_index` to avoid [this volar issue](https://github.com/vuejs/language-tools/issues/329)~
- try to make AppMap component more re-usable/generic
   - remove popup content. replace with slot and make popup content the responsibility of consuming/parent component.
   - make prop types generic instead of being tied to api functions
   - tweak feature info popup data handling
   - move "onMoveEnd" map handler to top level so its debouncing can be cancelled on unmount
   - remove "map.remove" call because it is already getting called before map mount (and calling it there [throws an error](https://github.com/Leaflet/Leaflet/issues/4806) for some reason... leaflet just doesn't play well with modern web frameworks)
   - for positioning feature labels, get info right from feature properties field
- tweak feature label appearance
- move tooltip that shows unit to just one place, the top left legend
- move feature popup content to SectionMap page component, with some typing (FeatureInfo)
- change logic of when certain stats (total/min/max/avg/etc) show or not
- add some util ts types

### CODE REVIEW

File: backend/app/models/cif_meta.py [hunk 1]
```diff
@@ -119,7 +119,7 @@
         },
         "PWS_Violations_Since_2016": {
             "label": "Public Water System Violations since 2016",
-            "unit": MeasureUnit.PERCENT,
+            "unit": MeasureUnit.COUNT,
```
Comment (vincerubinetti) [inline backend/app/models/cif_meta.py:122]: This seems to be an absolute count. Unless there is a division missing somewhere in the ingest.

File: frontend/src/pages/home/SectionMap.vue [hunk 2]
```diff
@@ -468,25 +566,50 @@ import {
...
   if (!values.value) return [];
   const value = values.value!;
-  const unit = value.unit;
+  const unit = value.unit ?? "";
+
   const stats: KeysOfValue<typeof values.value, number>[] = [];
-  if (["count", "dollar_amount"].includes(unit ?? "")) stats.push("total");
-  stats.push("min", "max", "avg", "median");
+
+  if (["count"].includes(unit)) stats.push("total");
```
Comment (vincerubinetti) [inline frontend/src/pages/home/SectionMap.vue:595]: Please take a look at this logic especially.

File: backend/app/models/cif_meta.py (hunk 1 above)
Comment (falquaddoomi) [reply backend/app/models/cif_meta.py:122]: No, you're right; it's supposed to be a count. Nice catch!

**Comment by vincerubinetti** [comment]
> I'd argue that any of the units could work with any of the stats except "total", which only makes sense for "count" units.

This is what I thought at first, but thinking about it more, I could see, for example, the "average" value being misleading for a % measure, because it's not really doing a global/state-wide average, but an average of each individual county percent, which can be different. Basically, the same issue as with "total".

I guess this is more a question for the whole group.

As for the implementation, I view min/max as tied to each other, as well as avg/median. I can't think of a case where we'd want one but not the other at the moment. I just pushed another commit which I think improves the readability.

**Pull request merged**